# Tarea grupal — IA en Producción — Grupo 5
## Punto 6 — Filtro por código de proyecto según año

**Integrante:** _(Cristina Andrea Corominas González)_
**Rama:** _(Cristina-Corominas)_

Este notebook es **autocontenido**: incluye su propia carga de librerías y del archivo
`plan_de_compras_2025.xlsx`, por lo que puede ejecutarse de forma independiente
(`Kernel > Restart & Run All`) sin depender de ningún otro notebook del equipo.


## 0. Carga de librerías y datos

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option('display.max_columns', 30)
plt.rcParams['figure.figsize'] = (10, 6)

In [ ]:
ARCHIVO = 'plan_de_compras_2025.xlsx'

df = pd.read_excel(ARCHIVO, sheet_name='PLAN DE COMPRAS 2025')
print('Dimensiones del dataset:', df.shape)
df.head()

Dimensiones del dataset: (831, 26)


,Unidad de Compra,ID Proyecto,Tipo Proyecto,Estado Proyecto,Código presupuestario,Nombre Proyecto,Descripción Proyecto,Cantidad de Ítems,Nombre ítem,Tipo compra,Cantidad Productos,Monto Unitario Ítem,Monto Total Ítem Año 2025,Cantidad OC,Meses envío OC,Nombre responsable,Cargo responsable,Teléfono responsable,Correo responsable,Fecha de Inicio Compra,Fecha Publicación PAC 2025,Cantidad OC Asociadas Ítem 2025,OC Asociada Item 2025,Item De Arraste,Monto De Arrastre,Meses envio OC de Arraste
0,Ministerio de Vivienda y Urbanismo(766),587-259-PC22,Proyecto estratégico,Publicado,2207-2209-33,Servicio arriendo de una plataforma de gestión...,Servicio de arriendo de una plataforma de gest...,2,Servicio arriendo de una plataforma de gestión...,1,1,125000000,20000000,3,"Abr 2023,Ene 2024,Ene 2025",María Teresa Zúñiga Silva,Coordinadora de Tecnologías Informáticas | Sis...,2 -2901 1198-,mzunigas@minvu.cl,2022-11-01,2022-11-08,0,NaN,1,20000000,2025-01-01
1,SEREMI MINVU V REGION,632-28-PC22,Proyecto operacional,Publicado,2206-2206001,Servicio de Mantención y Reparaciones Menores ...,Convenio de Suministro para el Servicio de Man...,1,Mantenimiento y Reparación de Edificaciones,1,1,20000000,4500000,5,"May 2022,Nov 2022,Ene 2023,Ene 2024,Ene 2025",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-02-01,2022-11-08,1,632-7-SE22,1,4500000,2025-01-01
2,SEREMI MINVU V REGION,632-32-PC22,Proyecto operacional,Publicado,2204001,Servicios de Impresión-Seremi Valparaiso,Servicios de Impresión de Materiales SEREMI y ...,1,Materiales de Oficina,1,1,15300000,3825000,4,"Oct 2022,Ene 2023,Ene 2024,Ene 2025",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-08-01,2022-11-08,0,NaN,1,3825000,2025-01-01
3,SEREMI MINVU V REGION,632-34-PC22,Proyecto operacional,Publicado,2209006,Servicio de Arriendo de Impresoras Multifuncio...,Servicio de Arriendo de Impresoras Multifuncio...,2,Arriendo de Equipos Informáticos,1,1,18000000,4500000,6,"Oct 2022,Nov 2022,Dic 2022,Ene 2023,Ene 2024,E...",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-08-01,2022-11-08,1,632-13-CC23,1,4500000,2025-01-01
4,SEREMI MINVU V REGION,632-41-PC22,Proyecto operacional,Publicado,2205005,Servicio de Telefonía Fija-Seremi Valparaiso,Servicio de Telefonía Fija SEREMI,2,Telefonía Fija,1,1,105600000,26400000,4,"Ene 2023,Ene 2024,Ene 2025,Ene 2026",Ariel Gardaix Gardaix,Encargado Sección Administración y Finanzas,322186802,agardaix@minvu.cl,2022-12-01,2022-11-08,0,NaN,1,26400000,2025-01-01


## 6. Filtro por código de proyecto según año

El `ID Proyecto` termina con un sufijo tipo `PC25`, `PC24`, `PC23`, etc., donde los **2 últimos dígitos**
indican el año del proyecto (ej. `PC25` → año 2025). Se extrae ese sufijo y se filtra el dataset para
dejar **solo los proyectos del año 25**.

In [ ]:
df['ID Proyecto'] = df['ID Proyecto'].astype('string').str.strip()
df['Sufijo Proyecto'] = df['ID Proyecto'].str.extract(r'([A-Za-z]{2}\d{2})$')
df['Año Proyecto'] = '20' + df['Sufijo Proyecto'].str[-2:]

df['Año Proyecto'].value_counts()

Año Proyecto
2025    648
2024     97
2023     81
2022      5
Name: count, dtype: Int64

In [ ]:
df_2025 = df[df['ID Proyecto'].str.endswith('PC25', na=False)].copy()

print(f"Registros totales: {len(df)}")
print(f"Registros año 2025 (PC25): {len(df_2025)}")

df_2025[['ID Proyecto', 'Sufijo Proyecto', 'Año Proyecto']].head(10)

Registros totales: 831
Registros año 2025 (PC25): 648


,ID Proyecto,Sufijo Proyecto,Año Proyecto
183,1583-4-PC25,PC25,2025
184,1583-4-PC25,PC25,2025
185,1583-4-PC25,PC25,2025
186,1583-5-PC25,PC25,2025
187,1583-5-PC25,PC25,2025
188,1583-5-PC25,PC25,2025
189,1583-5-PC25,PC25,2025
190,1583-6-PC25,PC25,2025
191,1583-8-PC25,PC25,2025
192,1583-8-PC25,PC25,2025


## Resumen

- **Filtro por año:** 648 de 831 registros corresponden a proyectos del año 2025 (`PC25`).
